GitUtils.ps1

```diff
@@ -129,56 +141,69 @@ function Get-GitStatus($gitDir = (Get-GitDirectory)) {
+                    'R' { $null = $indexModified.Add($matches['path1']); break }
+                    'C' { $null = $indexModified.Add($matches['path1']); break }
+                    'D' { $null = $indexDeleted.Add($matches['path1']); break }
+                    'U' { $null = $indexUnmerged.Add($matches['path1']); break }
+                }
+                switch ($matches['working']) {
+                    '?' { $null = $filesAdded.Add($matches['path1']); break }
+                    'A' { $null = $filesAdded.Add($matches['path1']); break }
+                    'M' { $null = $filesModified.Add($matches['path1']); break }
+                    'D' { $null = $filesDeleted.Add($matches['path1']); break }
+                    'U' { $null = $filesUnmerged.Add($matches['path1']); break }
+                }
+                continue
+            }
 
-                    '^## (?<branch>\S+?)(?:\.\.\.(?<upstream>\S+))?(?: \[(?:ahead (?<ahead>\d+))?(?:, )?(?:behind (?<behind>\d+))?\])?$' {
-                        $branch = $matches['branch']
-                        $upstream = $matches['upstream']
-                        $aheadBy = [int]$matches['ahead']
-                        $behindBy = [int]$matches['behind']
-                    }
+            '^## (?<branch>\S+?)(?:\.\.\.(?<upstream>\S+))?(?: \[(?:ahead (?<ahead>\d+))?(?:, )?(?:behind (?<behind>\d+))?\])?$' {
+                if ($sw) { dbg "Status: $_" $sw }
 
-                    '^## Initial commit on (?<branch>\S+)$' {
-                        $branch = $matches['branch']
-                    }
-                }
+                $branch = $matches['branch']
+                $upstream = $matches['upstream']
+                $aheadBy = [int]$matches['ahead']
+                $behindBy = [int]$matches['behind']
+                continue
+            }
+
+            '^## Initial commit on (?<branch>\S+)$' {
+                if ($sw) { dbg "Status: $_" $sw }
+
+                $branch = $matches['branch']
+                continue
             }
+
+            default { if ($sw) { dbg "Status: $_" $sw } }
+
         }
 
         if(!$branch) { $branch = Get-GitBranch $gitDir $sw }
+
         dbg 'Building status object' $sw
-        $indexPaths = $indexAdded + $indexModified + $indexDeleted + $indexUnmerged
-        $workingPaths = $filesAdded + $filesModified + $filesDeleted + $filesUnmerged
-        $index = New-Object PSObject @(,@($indexPaths | ?{ $_ } | Select -Unique)) |
-            Add-Member -PassThru NoteProperty Added    $indexAdded |
-            Add-Member -PassThru NoteProperty Modified $indexModified |
-            Add-Member -PassThru NoteProperty Deleted  $indexDeleted |
-            Add-Member -PassThru NoteProperty Unmerged $indexUnmerged
-        $working = New-Object PSObject @(,@($workingPaths | ?{ $_ } | Select -Unique)) |
+        #
+        # This collection is used twice, so create the array just once
+        $filesAdded = $filesAdded.ToArray()

```

Comment:
I didn't want to change anything, so I create a single array for the two places this is used. All the others are used just once, so inline was cleaner.